From 80dda456abdc9202fef387798b59e8fee2e7167e Mon Sep 17 00:00:00 2001
From: Lars Schmertmann <Lars.Schmertmann@governikus.de>
Date: Thu, 11 Jun 2020 09:06:18 +0200
Subject: Add NfcTagType4A and NfcTagType4B to QNearFieldTarget::Type

Let the applications know which signalling type is used, is
important for the support because a lot of mobile devices
have different behavior.

Task-number: QTBUG-81824
Change-Id: I8c3e2143d2f005c416173d7f62d04a2a272c616a
Reviewed-by: Alex Blasche <alexander.blasche@qt.io>
---
 src/nfc/qnearfieldtarget.cpp         | 5 ++++-
 src/nfc/qnearfieldtarget.h           | 2 ++
 src/nfc/qnearfieldtarget_android.cpp | 4 ++--
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git x/qtconnectivity/src/nfc/qnearfieldtarget.cpp y/qtconnectivity/src/nfc/qnearfieldtarget.cpp
index 850408fe..3d0edbe7 100644
--- x/qtconnectivity/src/nfc/qnearfieldtarget.cpp
+++ y/qtconnectivity/src/nfc/qnearfieldtarget.cpp
@@ -91,7 +91,10 @@ QT_BEGIN_NAMESPACE
     \value NfcTagType1      An NFC tag type 1 target.
     \value NfcTagType2      An NFC tag type 2 target.
     \value NfcTagType3      An NFC tag type 3 target.
-    \value NfcTagType4      An NFC tag type 4 target.
+    \value NfcTagType4      An NFC tag type 4 target. This value is used if the NfcTagType4
+                            cannot be further refined by NfcTagType4A or NfcTagType4B below.
+    \value NfcTagType4A     An NFC tag type 4 target based on ISO/IEC 14443-3A.
+    \value NfcTagType4B     An NFC tag type 4 target based on ISO/IEC 14443-3B.
     \value MifareTag        A Mifare target.
 */
 
diff --git x/qtconnectivity/src/nfc/qnearfieldtarget.h y/qtconnectivity/src/nfc/qnearfieldtarget.h
index 868b52d5..2f224379 100644
--- x/qtconnectivity/src/nfc/qnearfieldtarget.h
+++ y/qtconnectivity/src/nfc/qnearfieldtarget.h
@@ -70,6 +70,8 @@ public:
         NfcTagType2,
         NfcTagType3,
         NfcTagType4,
+        NfcTagType4A,
+        NfcTagType4B,
         MifareTag
     };
     Q_ENUM(Type)
diff --git x/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp y/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
index 78da6ac2..6b2bf7ff 100644
--- x/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
+++ y/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
@@ -449,10 +449,10 @@ QNearFieldTarget::Type NearFieldTarget::getTagType() const
         if ((sakS & 0x0064) == 0x0000)
             return NfcTagType2;
         else if ((sakS & 0x0064) == 0x0020)
-            return NfcTagType4;
+            return NfcTagType4A;
         return ProprietaryTag;
     } else if (m_techList.contains(NFCBTECHNOLOGY)) {
-        return NfcTagType4;
+        return NfcTagType4B;
     } else if (m_techList.contains(NFCFTECHNOLOGY)) {
         return NfcTagType3;
     }
-- 
2.27.0

